plugins/modules/site_intent.py

```diff
@@ -807,12 +807,15 @@ def get_diff_merged(self, config):
         else:
             # Creating New Site
             site_params = self.want.get("site_params")
-            if site_params['site']['building']:
-                building_details = {}
-                for key, value in site_params['site']['building'].items():
-                    if value is not None:
-                        building_details[key] = value
-                site_params['site']['building'] = building_details
+            try:
+                if site_params['site']['building']:
+                    building_details = {}
+                    for key, value in site_params['site']['building'].items():
+                        if value is not None:
+                            building_details[key] = value
+                    site_params['site']['building'] = building_details
+            except Exception as e:
+                self.log("Given site is not of type building so no need to remove None keys from the site_params dictionary", "INFO")
```

Comment:
"The site '{site_name}' is not of type building; hence, there is no need to filter out 'None' values from the 'site_params' dictionary." 
OR
"The site '{site_name}' is not categorized as a building; hence, there is no need to filter out 'None' values from the 'site_params' dictionary."

Reply:
added.